File: backend/internal/models/meal.go
package models

import (
	"time"
)

type Meal struct {
	ID            uint       `gorm:"primaryKey" json:"id"`
	Name          string     `json:"name"`
	Description   *string    `json:"description"`
	Calories      int        `json:"calories"`
	ProteinG      int        `json:"protein_g"`
	CarbsG        int        `json:"carbs_g"`
	FatG          int        `json:"fat_g"`
	MealType      *string    `json:"meal_type"`
	ImageUrl      *string    `json:"image_url"`
	DietaryTags   *string    `json:"dietary_tags"`
	MedicalTags   *string    `json:"medical_tags"`
	TargetWorkout *string    `json:"target_workout"`
	CreatedAt     time.Time  `json:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at"`
}

File: frontend/app/Models/Meal.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Meal extends Model
{
    protected $fillable = [
        'name', 'description', 'calories', 'protein_g', 'carbs_g', 'fat_g',
        'meal_type', 'image_url', 'dietary_tags', 'medical_tags', 'target_workout'
    ];

    protected function casts(): array
    {
        return [
            'dietary_tags' => 'array',
            'medical_tags' => 'array',
            'target_workout' => 'array',
        ];
    }
}
